packages/@sanity/base/src/preview/constants.tsx

```diff
@@ -0,0 +1,49 @@
+import React from 'react'
+import styled from 'styled-components'
+import {Flex, Text} from '@sanity/ui'
+import {WarningOutlineIcon, AccessDeniedIcon} from '@sanity/icons'
+import {PreparedValue} from './prepareForPreview'
+
+export const INCLUDE_FIELDS_QUERY = ['_id', '_rev', '_type']
+export const INCLUDE_FIELDS = [...INCLUDE_FIELDS_QUERY, '_key']
+
+// NOTE: have to use color inherit to make it work correctly with the
+// `isSelected` state in the list pane
+const IconSizer = styled(Text)`
+  color: inherit;
+`
+
+function IconWrapper({children}: {children: React.ReactNode}) {
+  return (
+    <Flex>
+      <IconSizer size={3}>{children}</IconSizer>
+    </Flex>
+  )
+}
+
+export class InsufficientPermissionsError extends Error {}
+
+export const INVALID_PREVIEW_FALLBACK: PreparedValue = {
+  // The `<small>` element is used for more compatibility
+  // with the different downstream preview components.
+  title: <small>Invalid preview config</small>,
+  subtitle: <small>Check the error log in the console</small>,
+  media: (
+    <IconWrapper>
+      <WarningOutlineIcon />
+    </IconWrapper>
+  ),
+  _meta: {type: 'invalid_preview'},
```

Comment:
Will this be exposed to custom preview components? If so, we might consider naming it `_internal` to make it clear that it's not meant to be relied upon?

Would it also be possible to use this to defer the rendering of the error states to the actual preview component? (having jsx in here has always felt a bit backwards to me)

Reply:
> Will this be exposed to custom preview components?

Yes. This one is kind of interesting so I think I'll give the full story for more context.

Previously the invalid preview fallback returned a unique symbol that we checked and used to fall back to this invalid preview value. Custom preview components most likely didn't cover this case but that was okay because this is an exception case.

With the addition of the insufficient permissions case, we now have a case that's not an exception. In order to stay compatible with custom previous components, we don't really want to return a symbol so these fallbacks were introduced and passed down at this level.

> Would it also be possible to use this to defer the rendering of the error states to the actual preview component?

I'm also not super happy with the JSX here but it seemed like the only way to get some sort of special styling at this level without having to touch every preview component. This is why I used the `<small>` element, for semantic meaning that this is a note + a semi-hack to get it to look visually different in all contexts.

Reply:
As far as the naming, I can change it to `_internal` if you prefer, though I'm also okay with leaving this convention as is. I would say it's valid for downstream preview components to use but if we're unsure we can name it `_internal`.

Let me know what you think!

Reply:
I think it makes sense to name it `_internal_meta` (or something) for now to buy us some time to figure out a better way of passing this  type of information to preview component implementors. Easier to rename it to something official if/when we decide there's a real user need for it than going the opposite way :)

Reply:
I tried `_internal_meta` but that was breaking our eslint camelcase rules, which is an error level rule so i went with `_internalMeta`